lib/src/posthog_flutter_io.dart

```diff
@@ -1,17 +1,131 @@
+import 'dart:async';
+
 import 'util/platform_io_stub.dart'
     if (dart.library.io) 'util/platform_io_real.dart';
 
 import 'package:flutter/services.dart';
+
+import 'package:posthog_flutter/src/surveys/survey_service.dart';
 import 'package:posthog_flutter/src/util/logging.dart';
+import 'surveys/models/posthog_display_survey.dart' as models;
+import 'surveys/models/survey_callbacks.dart';
 
 import 'posthog_config.dart';
 import 'posthog_flutter_platform_interface.dart';
 
 /// An implementation of [PosthogFlutterPlatformInterface] that uses method channels.
 class PosthogFlutterIO extends PosthogFlutterPlatformInterface {
+  PosthogFlutterIO() {
+    _methodChannel.setMethodCallHandler(_handleMethodCall);
+  }
+
   /// The method channel used to interact with the native platform.
   final _methodChannel = const MethodChannel('posthog_flutter');
 
+  Future<dynamic> _handleMethodCall(MethodCall call) async {
```

Comment:
what will happen when we call those methods on Android and Web since its iOS only? we have to degrade gracefully with logging message instead

Reply:
I'd run a test on android and web with surveys enabled just to make sure we are not crashing peoples apps but rather degrading gracefully

Reply:
if its tested, sure